### PR TITLE
TSDK-366 - Ruby SDK Hosted Authentication missing optional parameters

### PR DESCRIPTION
**Pull request by kraju3**
# Description
Hosted Authentication method was missing the provider and redirect_on_error params. I have added them and created a simple test
JIRA --> https://nylas.atlassian.net/browse/TSDK-366

# License
I confirm that this contribution is made under the terms of the MIT license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Review by mrashed-dev — CHANGES_REQUESTED** [review]
@kraju3 Thanks for submitting this PR! The test is missing closing `end` keywords, please ensure that the test is runnable and passes correctly. Run `bundle exec rspec spec` to run the test suite.

Also you have style issues with your code, please run `bundle exec rubocop --config .rubocop.yml -a` and fix the styling issues.

Doing both will ensure the code is up to the codebase standards, and will pass the CI actions. Thanks!

File: spec/nylas/api_spec.rb [hunk 1]
```diff
@@ -71,6 +71,30 @@
...
+        "?client_id=2454354"\
+        "&redirect_uri=http%3A%2F%2Fexample.com"\
+        "&response_type=code"\
+        "&login_hint=email%40example.com"\
+        "&state=some-state"\
+        "&scopes=email%2Ccalendar"\
+        "provider=gmail"\
+        "redirect_on_error=true"
+
+        expect(hosted_auth_url).to eq expected_url
```
[the remark this answers is not in the document]
Comment (kraju3) [reply spec/nylas/api_spec.rb:97]: Oh man! Thanks for the heads up lol

**Review by mrashed-dev — APPROVED** [review]
Amazing, thanks a lot @kraju3 !

**Comment by kraju3** [comment]
Anything to help! @mrashed-dev

**Pull request merged**